Floofbot/Modules/Fun.cs

```diff
@@ -168,5 +168,172 @@ public async Task RequestCatFact()
             }
             await Context.Channel.SendMessageAsync(fact);
         }
+
+        [Command("minesweeper")]
+        [Summary("Minesweeper minigame")]
+        public async Task Minesweeper([Summary("grid x")]int gridx, [Summary("grid y")]int gridy, [Summary("bomb count")]int bombs)
+        {
+           
+            //no negative numbers!
+            if(gridx < 1 || gridy < 1 || bombs < 0)
+            {
+                await Context.Channel.SendMessageAsync("Invalid grid size or bomb count");
+                return;
+            }
+            
+            //limits the size of the board
+            if(gridx > 10 || gridy > 10)
+            {
+                await Context.Channel.SendMessageAsync("Max Grid Size: 10 x 10");
+                return;
+            }
+            else if(bombs >= gridy * gridx)
+            {
+                await Context.Channel.SendMessageAsync("Too many bombs!");
+                return;
+            }
+
+            Gameboard game = new Gameboard(gridx, gridy, bombs);
+            EmbedBuilder builder = new EmbedBuilder();
+            builder.Title = ":bomb: Minesweeper";
+            builder.Color = EMBED_COLOR;
+            builder.Description = game.getBoard();
+
+            await Context.Channel.SendMessageAsync("", false, builder.Build());
+        }
+
+        public class Gameboard
+        {
+            private string[,] grid;
+            private int x;
+            private int y;
```

Comment:
x and y usually refer to a specific point
pls rename these vars width and height